src/runtime/TemporalObject.h

```diff
@@ -148,6 +148,18 @@ class TemporalObject : public Temporal {
 
     explicit TemporalObject(ExecutionState& state);
     explicit TemporalObject(ExecutionState& state, Object* proto);
+    virtual bool hasCalendar() const
+    {
+        return false;
+    }
+    bool isTemporalObject() const
+    {
+        return true;
+    }
+    virtual const TemporalCalendarObject* getCalendar() const
```

Comment:
Why did you define this function as const?
It seems meaningless because other parts calling this function cast the return value using const_cast.